Add info about TBBBind and custom allocators to known limitations (#1666)
uxlfoundation/oneTBB: doc/main/intro/limitations.rst

**Limitation:** oneTBB does not support ``fork()``. 

**Solution:** To work-around the issue, consider using ``task_scheduler_handle`` to join oneTBB worker threads before using ``fork()``.

Custom Allocator and Topology API Incompatibilities
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> **Limitation:** On Linux* OS, using custom memory allocators together with ``tbb::info`` and ``tbb::task_arena::constraints`` APIs may result in undefined behavior.

Reviewer: ```suggestion
**Limitation:** On Linux* OS, using custom memory allocators with ``tbb::info`` and ``tbb::task_arena::constraints`` APIs may result in undefined behavior.
```

> **Limitation:** On Linux* OS, using custom memory allocators together with ``tbb::info`` and ``tbb::task_arena::constraints`` APIs may result in undefined behavior.

Reviewer 2: I don't like the formulation `custom memory allocators` here since it can be interpreted as usage of custom allocator types. Since this issue only relates to the malloc replacement libraries like tbbmalloc, jemalloc, etc., I propose to use "dynamic malloc replacement" or "dynamic allocating functions replacement" instead.

Reviewer 2: Also, do we use "undefined behavior" when defining the known limitations? It seems like we can be more concrete and say it is runtime failures.

> **Solution:** Set ``TBB_ENABLE_SANITIZERS=1`` in the environment to hint that custom allocators are used.

Reviewer: ```suggestion
**Solution:** Set ``TBB_ENABLE_SANITIZERS=1`` in the environment. This informs that custom allocators are used.
```

> **Solution:** Set ``TBB_ENABLE_SANITIZERS=1`` in the environment to hint that custom allocators are used.

Reviewer 2: It does not seem like the `TBB_ENABLE_SANITIZERS` variable is about usage of custom dynamic malloc replacements and also since the deepbind would be dropped, can we guarantee that the tbbbind will work correct in all use-cases? 
Does it makes sense to provide `LD_PRELOAD`-based workaround instead?

Author: I agree that the environment variable name seems unconnected. However, the `LD_PRELOAD`-based workaround might cause more issues due to incorrect/incompatible TBBBind/HWLOC versions.

When deepbind is dropped, I believe we would remove the known limitation and the environment variable, because the globally-defined malloc replacement would always be used.
